src/reducers/viewconfig.ts

```diff
@@ -124,39 +125,74 @@ const normalizeTree = (tree: ComponentConfig[], collectionId: string, startIndex
     return flatTree;
 };
 
-const denormalizeNode = (item: NormalizedComponentConfig, state: ViewConfigReducer): ComponentConfig => {
-    item = { ...item };
+const denormalizePath = (childNode: NormalizedComponentConfig): NormalizedComponentConfig => {
+    childNode = { ...childNode };
 
-    if (item.childIds && !!item.childIds.length) {
-        item.subComponents = [];
+    if (childNode.referencePath) {
+        childNode.value = mendPath(childNode.referencePath);
+        delete childNode.referencePath;
+    }
+
+    return childNode;
+};
 
+const denormalizeNode = (item: NormalizedComponentConfig, state: ViewConfigReducer): ComponentConfig | string => {
+    item = denormalizePath(item);
+    item.subComponents = [];
+
+    if (item.childIds && !!item.childIds.length) {
         for (const child of item.childIds) {
             const childNode = getNodeById(child, state);
 
             if (childNode) {
+                const error = componentErrors(childNode);
+                if (error) {
+                    return error;
+                }
+
                 const denormalizedChildNode = denormalizeNode(childNode, state);
 
-                if (denormalizedChildNode) {
+                if (denormalizedChildNode && typeof denormalizedChildNode !== 'string') {
                     item.subComponents.push(denormalizedChildNode);
                 }
             }
         }
     }
 
+    if (item.__typename) {
+        delete item.__typename;
+    }
+
     delete item.id;
     delete item.childIds;
     delete item.name;
     return item;
 };
 
-export const denormalizeTree = (state: ViewConfigReducer): ComponentConfig[] | void => {
+export const denormalizeTree = (state: ViewConfigReducer): ComponentConfig[] | string => {
     let denormalizedTree: ComponentConfig[] = [];
 
-    for (const rootNodeIdx of state[0].childIds) {
+    const parentNode = getNodeById(0, state);
+
+    if (!parentNode || !parentNode.childIds) {
+        return 'no parentnode!!';
```

Comment:
shouldn't this throw instead?

Reply:
Deleting it instead, as there is always a node with id 0, even without data